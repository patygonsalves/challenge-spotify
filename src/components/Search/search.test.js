import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import userEvent from '@testing-library/user-event'

import { render, screen } from '@testing-library/react'

import SearchField from '.'

describe('<SearchField />', () => {
  test('should render SearchField', () => {
    const { container } = render(
      <SearchField
        name={'search'}
        value={''}
        placeholder={'busque seu artista...'}
        onChange={jest.fn()}
      />
    )

    expect(screen.getByPlaceholderText(/busque seu artista.../i)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  test('calls change handler for an input', () => {
    const handleChange = jest.fn()
    const { getByRole } = render(
      <SearchField
        name={'search'}
        value={''}
        placeholder={'busque seu artista...'}
        onChange={handleChange}
      />
    )
    const input = getByRole('input')
    userEvent.type(input, 'taylor swift')
    expect(handleChange).toHaveBeenCalled()
  })
})
