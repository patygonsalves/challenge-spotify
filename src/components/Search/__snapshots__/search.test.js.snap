// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchField /> should render SearchField 1`] = `
<div
  class="inputX-field"
>
  <input
    class="inputX-input"
    name="search"
    placeholder="busque seu artista..."
    role="input"
    spellcheck="false"
    value=""
  />
</div>
`;
